File: python/Mahasiswa.py
#import data dari file siak
from Siak import Siak
#Mahasiswa yang merupakan anak dari Siak
class Mahasiswa(Siak):
    
    #constructor
    def __init__(self, nim = 0, fakultas = "", nama ="", nik = 0, jenisKelamin = "", asalUniv = "", emailEdu = ""):
        #untuk mengakses variabel dan metode pada parent
        super().__init__(asalUniv, emailEdu, nik, nama, jenisKelamin)
        self.nim = nim
        self.fakultas = fakultas
        
    #setter nim dan fakultas mahasiswa
    def setNim(self, nim):
        self.nim = nim
    
    def setFakultas(self, fakultas):
        self.fakultas = fakultas
        
    #getter untuk nim dan fakultas mahasiswa
    def getNim(self):
        return self.nim
    
    def getFakultas(self):
        return self.fakultas

File: c++/Program/Mahasiswa.cpp
#include"Header.hh"

Mahasiswa::Mahasiswa() : Siak(){
}

Mahasiswa::Mahasiswa(int nik, string nama, string jenis_kelamin, string asal_univ, string email_edu) : Siak(nik, nama, jenis_kelamin, asal_univ, email_edu) {
}

Mahasiswa::Mahasiswa(int nik, string nama, string jenis_kelamin, string asal_univ, string email_edu, int nim, string fakultas) : Siak(nik, nama, jenis_kelamin, asal_univ, email_edu){
    this->nim = nim;
    this->fakultas = fakultas;
}

void Mahasiswa::setNim(int nim){
    this->nim = nim;
}
int Mahasiswa::getNim(){
    return this->nim;
}

void Mahasiswa::setFakultas(string fakultas){
    this->fakultas = fakultas;
}
string Mahasiswa::getFakultas(){
    return this->fakultas;
}

Mahasiswa::~Mahasiswa(){
}